openmc/filter.py

```diff
@@ -132,18 +135,18 @@ def bins(self, bins):
         if self._type in ['cell', 'cellborn', 'surface', 'material',
                           'universe', 'distribcell']:
             for edge in bins:
-                if not is_integer(edge):
+                if not isinstance(edge, Integral):
                     msg = 'Unable to add bin "{0}" to a {1} Filter since ' \
-                          'it is a non-integer'.format(edge, self._type)
+                          'it is not an integer'.format(edge, self._type)
                     raise ValueError(msg)
                 elif edge < 0:
                     msg = 'Unable to add bin "{0}" to a {1} Filter since ' \
-                          'it is a negative integer'.format(edge, self._type)
+                          'it is negative'.format(edge, self._type)
```

Comment:
It might be useful to have a ``check_less_than(...)`` and a ``check_greater_than(...)`` as well